### PR TITLE
Update dependencies and prepare script

### PR DESCRIPTION
**Pull request by zFernand0**
**What It Does**

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [ ] added/updated automated tests
- [ ] updated the changelog
- [ ] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,44 +41,44 @@
...
-    "prepare": "husky install"
+    "prepare": "husky"
   },
   "imperative": {
     "configurationModule": "lib/pluginDef.js"
   },
   "dependencies": {
-    "@zowe/core-for-zowe-sdk": "^7.23.2",
-    "@zowe/zosmf-for-zowe-sdk": "^7.23.2"
+    "@zowe/core-for-zowe-sdk": "^7.29.2",
```
Comment (gejohnston) [inline package.json:50]: Should our sample remain a V2 sample, or should it be a V3 sample?

File: __tests__/__integration__/cli/list/cli.list.profile-args.integration.test.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 import * as fs from "fs";
 import * as path from "path";
-import { ITestEnvironment, TestEnvironment, runCliScript, isStderrEmptyForProfilesCommand } from "@zowe/cli-test-utils";
+import { ITestEnvironment, TestEnvironment, runCliScript, isStderrEmptyForProfilesCommand, stripProfileDeprecationMessages } from "@zowe/cli-test-utils";
```
[the remark this answers is not in the document]
Comment (traeok) [reply __tests__/__integration__/cli/list/cli.list.profile-args.integration.test.ts:13]: related: `isStderrEmptyForProfilesCommand` can be removed from imports

File: package.json (hunk 1 above)
Comment (zFernand0) [reply package.json:50]: I don't mind moving things into a zowe-v2-lts branch after this PR is merged 🙏

File: .husky/commit-msg [hunk 3]
```diff
@@ -1,6 +1,4 @@
 #!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply .husky/commit-msg:2]: After a husky 9.1.6 update, they issue a warning message saying that having that will break in husky 10.0.0.

And it seems to still work without it 😋 

image

File: .husky/commit-msg [hunk 4]
```diff
@@ -1,6 +1,4 @@
...
-
 # If not a merge commit, require commit to be signed off
 if [ ! -e .git/MERGE_MSG ] && ! grep -q "^Signed-off-by: " "$1"; then
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply .husky/commit-msg]: resolved by 241ceea80ed71c7a073ed6e0e57ef862003e7d92

File: __tests__/__integration__/cli/list/cli.list.profile-args.integration.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zFernand0) [reply __tests__/__integration__/cli/list/cli.list.profile-args.integration.test.ts]: resolved by 241ceea80ed71c7a073ed6e0e57ef862003e7d92

File: .husky/commit-msg (hunk 3 above)
[the remark this answers is not in the document]
Comment (zFernand0) [reply .husky/commit-msg:2]: removed the shebang in 241ceea80ed71c7a073ed6e0e57ef862003e7d92

**Comment by zFernand0** [comment]
waiting for a new version from:
- https://github.com/zowe/zowe-cli/pull/2329

**Review by gejohnston — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @zFernand0!

**Pull request merged**